### ISSUE TITLE
SNOW-777893:  Cannot access member "fetchone" for type "dict[str, Any]"

### ISSUE THREAD

**Issue by tekumara** [snowflakedb/snowflake-connector-python#1498]
Given 
```python
    with conn.cursor() as cur:
        results = cur.execute(count_query)
        if results:
            row = results.fetchone()
```

I now get the following vscode/pyright type error on line 4:
```
 error: Cannot access member "fetchone" for type "dict[str, Any]"
```
Or mypy:
```
error: Item "Dict[str, Any]" of "Union[SnowflakeCursor, Dict[str, Any]]" has no attribute "fetchone"  [union-attr]
```

Because of the type signature:
```
    def execute(
        self,
...
    ) -> Self | dict[str, Any] | None:
```

This was introduced in 3.0.1 and still present in 3.0.2

**Comment by sfc-gh-spanaite** [comment]
Hi @tekumara and thanks for reporting this. I am able to reproduce using `mypy` and your sample code, I also get:

```python
$ mypy test.py
test.py:41: error: Item "Dict[str, Any]" of "Union[SnowflakeCursor, Dict[str, Any]]" has no attribute "fetchone"  [union-attr]
Found 1 error in 1 file (checked 1 source file)
(test) $
```

**Comment by sfc-gh-aling** [comment]
thanks @tekumara , I can reproduce the issue +1. will have a PR out to address it

**Comment by tekumara** [comment]
Thank you! I've raised a related issue here #1502

**Comment by sfc-gh-stan** [comment]
Closing since the fix #1501 is merged.
